main.go

```diff
@@ -428,15 +454,18 @@ func installOption(listAll bool, custBinPath, mirrorURL *string) {
 
 // install when tf file is provided
 func installTFProvidedModule(dir string, custBinPath, mirrorURL *string) {
-	fmt.Printf("Reading required version from terraform file\n")
+	fmt.Println("Reading required version from terraform file")
 	module, _ := tfconfig.LoadModule(dir)
-	tfconstraint := module.RequiredCore[0] //we skip duplicated definitions and use only first one
+	if len(module.RequiredCore) >= 1 {
+		fmt.Println("The provided directory does not have required terraform version constraint.")
+		os.Exit(1)
+	}
+	tfconstraint := strings.Join(module.RequiredCore, ", ")
```

Comment:
This is the change into existing logic which may change UX.

Since I'm not aware of why @warrensbox initially decided to pick only first TF version constraint and ignore all consequent definitions, we might want to implement this the way Terraform does this. Is `strings.Join(module.RequiredCore, ", ")` what TF does? Could you please point me to TF code where this is done. Thanks.

Reply:
I'd say it is a way to check all of them. An example of doing that one by one:
https://github.com/hashicorp/terraform-schema/blob/main/earlydecoder/decoder.go#L30-L42
```
	var coreRequirements version.Constraints
	for _, rc := range mod.RequiredCore {
		c, err := version.NewConstraint(rc)
		if err != nil {
			diags = append(diags, &hcl.Diagnostic{
				Severity: hcl.DiagError,
				Summary:  "Unable to parse terraform requirements",
				Detail:   fmt.Sprintf("Constraint %q is not a valid constraint: %s", rc, err),
			})
			continue
		}
		coreRequirements = append(coreRequirements, c...)
	}
```

Reply:
What I mean is changing "usual" way of "detecting" required TF version just like that may cause unpredictable and unexpected behavior on user end. So I'd suggest to split this off into a dedicated PR **and** discuss whether we want to go with `strings.Join()` like you implemented this in this PR, or e.g. like this is implemented in the example above that you provided so that we don't introduce custom way of determining required TF version.

@warrensbox @jukie FYI

Reply:
I'd say that taking into account all available constraints is a proper way to do that, but I've reverted that change - it is a kind of an edge case and may be discussed separately